lib/mayaUsd/nodes/proxyShapeBase.cpp

```diff
@@ -496,9 +496,8 @@ MayaUsdProxyShapeBase::computeInStageDataCached(MDataBlock& dataBlock)
 
             usdStage->SetEditTarget(usdStage->GetSessionLayer());
         }
-        else if (!usdStage) {
+        else {
             // Create a new stage in memory with an anonymous root layer.
-            UsdStageCacheContext ctx(UsdMayaStageCache::Get());
```

Comment:
I saw a question from @seando-adsk in my notifications but now I can't find it on GitHub.

It is true that currently we may not have any workflow which would require this stage to be inserted into the cache. At the same time, I don't see what could go wrong if we do insert it and keep the code symmetric with the case where we open the stage from file.

Reply:
I think you couldn't find it because the conversation got resolved. Here is the link:

https://github.com/Autodesk/maya-usd/pull/519#discussion_r429422220

I unresolved that conversation so lets continue talking there.